src/Tizen.Network.Smartcard/Tizen.Network.Smartcard/SmartcardManagerImpl.cs

```diff
@@ -106,16 +101,22 @@ internal IEnumerable<SmartcardReader> GetReaders()
                 SmartcardErrorFactory.ThrowSmartcardException(ret);
             }
 
+            IntPtr tempPtr = readerPtr;
             for (int i = 0; i < len; i++)
             {
-                int readerID = Marshal.ReadInt32(readerPtr);
+                int readerID = Marshal.ReadInt32(tempPtr);
 
                 SmartcardReader readerItem = new SmartcardReader(readerID);
-                _readerList.Add(readerItem);
-                readerPtr += sizeof(int);
+                readerList.Add(readerItem);
+                tempPtr += sizeof(int);
             }
 
-            return _readerList;
+            if (len > 0)
+            {
+                Marshal.FreeHGlobal(readerPtr);
+            }
```

Comment:
This is wrong.
You should call free from libc.

Refer to https://github.com/Samsung/TizenFX/blob/master/src/ElmSharp/Interop/Interop.Libc.cs#L24

Reply:
I change the code. Please check it.